src/cpp/lpnamer/input_reader/WeightsFileWriter.cpp

```diff
@@ -30,19 +27,17 @@ void YearlyWeightsWriter::CreateWeightFile() {
 
 void YearlyWeightsWriter::FillMpsWeightsMap() {
   mps_weights_.clear();
-  auto zip_reader = ArchiveReader(antares_archive_path_);
-  zip_reader.Open();
-  zip_reader.LoadEntriesPath();
-  const auto& mps_files = zip_reader.GetEntriesPathWithExtension(".mps");
+  std::vector<std::filesystem::path> mps_files;
+  for (auto& p : std::filesystem::directory_iterator(xpansion_lp_dir_)) {
+    if (p.path().extension() == ".mps") mps_files.emplace_back(p.path());
```

Comment:
Please use braces in the “if” condition to keep the same format/style as the code.